### Задача 1
Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

Для расчета будем использовать весь промежуток, т.е. [200, 800]

Рассчитаем среднее значение и дисперсию

In [1]:
a = 200
b = 800
avg = (a+b)/2
avg


500.0

In [2]:
Dx = pow((b-a),2)/12
Dx

30000.0

### Задача 2
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
Если да, найдите ее.

Данная задача обратна предыдущей

In [3]:
a = 0.5
Dx = 0.2


Выразим b по формуле дисперсии

$$D(x) = ((b-a)^2)/12$$
$$D(x) = (b^2-2ba+a^2)/12$$
$$b^2-2ba+a^2-12D(x) = 0$$
$$b^2-(2*0.5)b+0.25-(12*0.2) = 0$$

Найдем дискриминант

In [4]:
aa = 1
bb = -2*0.5
cc = pow(0.5,2) - (12*0.2)
print(aa,bb,cc)

1 -1.0 -2.1500000000000004


In [5]:
import math

discr = bb ** 2 - 4 * aa * cc
print("Дискриминант D = %.2f" % discr)
 
if discr > 0:
    x1 = (-bb + math.sqrt(discr)) / (2 * aa)
    x2 = (-bb - math.sqrt(discr)) / (2 * aa)
    print("x1 = %.2f \nx2 = %.2f" % (x1, x2))
elif discr == 0:
    x = -bb / (2 * aa)
    print("x = %.2f" % x)
else:
    print("Корней нет")

Дискриминант D = 9.60
x1 = 2.05 
x2 = -1.05


Проверим данные границы

In [6]:
Dx1 = (pow((x1-a),2))/12
Dx1

0.20000000000000004

In [7]:
Dx2 = (pow((x2-a),2))/12
Dx2

0.20000000000000004

In [8]:
Dx2 = (pow((x2-a),2))/12
Dx2

0.20000000000000004

Второй корень нам не подходит т.к. -1.05 < 0.5. Следовательно, правая граница равна 2.05<br>
Найдем математическое ожидание:

In [9]:
b = x1
Mx = (b+a)/2
Mx

1.2745966692414834

### Задача 3
Непрерывная случайная величина X распределена нормально и задана плотностью распределения
$$f(x) = (1 / (4 * sqrt(2*pi))) * (exp(-(x+2)**2) / 32)$$
Найдите:
а). M(X)
б). D(X)
в). std(X) (среднее квадратичное отклонение)

Для данного выражения M(x) будет равно а. а = -2<br>
D(x) = sigma^2, тогда D(x) = 32 / 2 = 16<br>
Std = sigma, тогда std = sqrt(16) = 4 <br>

### Задача 4
Рост взрослого населения города X имеет нормальное распределение.
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.
Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:<br>
а). больше 182 см<br>
б). больше 190 см<br>
в). от 166 см до 190 см<br>
г). от 166 см до 182 см<br>
д). от 158 см до 190 см<br>
е). не выше 150 см или не ниже 190 см<br>
ё). не выше 150 см или не ниже 198 см<br>
ж). ниже 166 см.<br>

Воспользуемся функцией norm.cdf(sigma)<br>
где sigma - среднеквадратичное отклонение

In [12]:
from scipy.stats import norm

Для подзадачи а

In [11]:
Z = 1 - norm.cdf(1)
Z

0.15865525393145707

Для подзадачи б

In [13]:
Z = 1 - norm.cdf(2)
Z

0.02275013194817921

Для подзадачи в

In [16]:
Z1 = norm.cdf(-1)
Z2 = norm.cdf(2)
Z = Z2 - Z1
Z

0.8185946141203637

Для подзадачи г

In [17]:
Z1 = norm.cdf(-1)
Z2 = norm.cdf(1)
Z = Z2 - Z1
Z

0.6826894921370859

Для подзадачи д

In [18]:
Z1 = norm.cdf(-2)
Z2 = norm.cdf(2)
Z = Z2 - Z1
Z

0.9544997361036416

Для подзадачи е

In [19]:
Z1 = norm.cdf(-3)
Z2 = 1- norm.cdf(2)
Z = Z2 - Z1
Z

0.021400233916549116

Для подзадачи ё

In [20]:
Z1 = norm.cdf(-3)
Z2 = 1- norm.cdf(3)
Z = Z2 - Z1
Z

1.0191500421363742e-17

In [21]:
### Вне диапазона?

Для подзадачи ж

In [21]:
Z = norm.cdf(1)
Z

0.8413447460685429

### Задача 5
На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

Т.к. sigma = sqrt(D(x)), то sigma = 5

In [23]:
kolvo = (190 - 178) / 5
kolvo

2.4